packages/snaps-rpc-methods/src/restricted/manageState.ts

```diff
@@ -267,17 +260,14 @@ export function getManageStateImplementation({
   decrypt,
 }: ManageStateMethodHooks) {
   return async function manageState(
-    options: RestrictedMethodOptions<ManageStateArgs>,
-  ): Promise<null | Record<string, Json>> {
+    options: RestrictedMethodOptions<ManageStateParams>,
+  ): Promise<ManageStateResult> {
     const {
       params = {},
       method,
       context: { origin },
     } = options;
-    const { operation, newState, encrypted } = getValidatedParams(
-      params,
-      method,
-    );
+    const { operation, encrypted } = getValidatedParams(params, method);
```

Comment:
Curious around this change, should we not have `newState` exposed here? Feels slightly risky to just use the state raw from params later

Reply:
TypeScript can't infer the type of `newState` at this point. I've refactored it to use the result of `getValidatedParams`.